tests/executors/test_local_executor.py

```diff
@@ -55,14 +55,16 @@ def execution_parallelism(self, parallelism=0):
         else:
             executor.end()
 
-        for i in range(self.TEST_SUCCESS_COMMANDS):
-            key = success_key.format(i)
-            self.assertTrue(executor.event_buffer[key], State.SUCCESS)
-        self.assertTrue(executor.event_buffer['fail'], State.FAILED)
+        if isinstance(executor.impl, LocalExecutor._LimitedParallelism):
```

Comment:
Could you please describe what problem we are actually solving here (either in the commit message or in the JIRA issue). I am not sure what is the reasoning for that change and having some context will be useful to understand if it works as intended.

Reply:
So when `executor.end()`is called, it ensures that the queues are empty with:
https://github.com/apache/airflow/blob/master/airflow/executors/local_executor.py#L195-L196
https://github.com/apache/airflow/blob/master/airflow/executors/local_executor.py#L205

So I am adding assertion in the test to ensure that.

Reply:
Also I have updated the jira.